p74/src/bin/rustcrypto_sha2.rs

```diff
@@ -0,0 +1,69 @@
+use sha2::{Digest, Sha224, Sha256, Sha384, Sha512, Sha512_224, Sha512_256};
+
+fn sha_224(input_string: &str) -> String {
+    let mut hasher = Sha224::new();
+    hasher.update(input_string.as_bytes());
+    let result = hasher.finalize();
+    format!("{:x}", result)
+}
+
+fn sha_256(input_string: &str) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(input_string.as_bytes());
+    let result = hasher.finalize();
+    format!("{:x}", result)
+}
+
+fn sha_384(input_string: &str) -> String {
+    let mut hasher = Sha384::new();
+    hasher.update(input_string.as_bytes());
+    let result = hasher.finalize();
+    format!("{:x}", result)
+}
+
+fn sha_512(input_string: &str) -> String {
+    let mut hasher = Sha512::new();
+    hasher.update(input_string.as_bytes());
+    let result = hasher.finalize();
+    format!("{:x}", result)
+}
+
+fn sha_512_224(input_string: &str) -> String {
+    let mut hasher = Sha512_224::new();
+    hasher.update(input_string.as_bytes());
+    let result = hasher.finalize();
```

Comment:
You can use `Sha256::digest(input_string)` to reduce number of lines. It also accepts all types which implement `AsRef<[u8]>`, so you even can remove the explicit `as_bytes` call.